include/multipass/yaml_node_utils.h

```diff
@@ -48,6 +48,10 @@ YAML::Node make_cloud_init_network_config(const std::string& default_mac_addr,
 YAML::Node add_extra_interface_to_network_config(const std::string& default_mac_addr,
                                                  const NetworkInterface& extra_interface,
                                                  const std::string& network_config_file_content);
+// the make_cloud_init_network_config and add_extra_interface_to_network_config functions are adapted to generate the
+// new format network-config file (having default interface present and having dhcp-identifier: mac on every network
+// interface). At the same time, it also needs to take care of the pre-existed file, meaning that the generated file
+// from the old file should have the new format.
```

Comment:
Could you please clarify the intention with this comment? I am confused by the placement and the content. What do you mean by "are adapted"? Are you referring to this PR? Because this would stay in the code from now on. I am honestly having a hard time extracting useful information from this, I wonder if we should just remove it.

Reply:
The changes of functions `make_cloud_init_network_config` and `add_extra_interface_to_network_config` are based on our discussion about treating the VirtualBox same IP address issue. The fix is to let the extra interface part of the network-config file has the new field `dhcp-identifier: mac` which forces the DHCP to generate the IP address based on mac addresses as opposed to other ids. However, if we make this change to extra interfaces, then we might want to do the same thing to the default interface to keep the consistency. But default interface was absent when there was no extra interfaces beforehand. 

That is why we have the new format network-config file and the bracketed part explains what that is. Besides, we need to take care pre-existed file which does not have the new format. 

Because of these reasons, we have these two functions that are centric to takeingthe default mac address and all extra interface info and re-create the network-config node. 

I think the comment here is useful because I was able to quickly recall things by reading it and link that to the implementation of the two functions. But your feedback is also valuable, feel free to suggest how to improve the clarity.

Reply:
As discussed, just short explanation that this needs to remain backward compatible with absent network-config/default interface. And move to relevant code definition.

Reply:
Done